lock-api/src/main/java/com/palantir/lock/watch/TimestampToVersionMap.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.immutables.value.Value;
+
+import com.google.common.collect.Ordering;
+import com.google.common.collect.TreeMultimap;
+import com.palantir.lock.v2.LockToken;
+
+final class TimestampToVersionMap {
```

Comment:
I would consider tests for this, for anything that's hard to test in your integration test. I haven't read through that yet, so unclear if this comment is useful.

Reply:
Done.